File: play-interop-test-scala/src/main/scala/controllers/MyController.scala
/**
 * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
 */
package controllers

import akka.actor.ActorSystem
import akka.stream.Materializer
import example.myapp.helloworld.grpc.helloworld.{ GreeterServiceClient, HelloRequest }
import javax.inject.{ Inject, Singleton }
import play.api.mvc.{ AbstractController, ControllerComponents }

import scala.concurrent.ExecutionContext

// #using-client
@Singleton
class MyController @Inject() (implicit greeterClient: GreeterServiceClient, cc: ControllerComponents, mat: Materializer, exec: ExecutionContext) extends AbstractController(cc) {

  def sayHello(name: String) = Action.async { implicit request =>
    greeterClient.sayHello(HelloRequest(name))
      .map { reply =>
        Ok(s"response: ${reply.message}")
      }
  }

}
// #using-client

File: play-interop-test-java/src/main/java/controllers/MyController.java
/**
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package controllers;

import example.myapp.helloworld.grpc.GreeterServiceClient;
import example.myapp.helloworld.grpc.HelloRequest;
import play.mvc.Controller;
import play.mvc.Result;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.concurrent.CompletionStage;

// #using-client
@Singleton
public class MyController extends Controller {

  private final GreeterServiceClient greeterServiceClient;

  @Inject
  public MyController(GreeterServiceClient greeterServiceClient) {
    this.greeterServiceClient = greeterServiceClient;
  }

  public CompletionStage<Result> sayHello(String name) {
    return greeterServiceClient.sayHello(
        HelloRequest.newBuilder()
            .setName(name)
            .build()
    ).thenApply(response ->
      ok("response: " + response.getMessage())
    );
  };

}
// #using-client